### ISSUE TITLE
CrossEntropyLoss, ignore_index does not prevent back-prop if the logits are -inf

### ISSUE THREAD

**Issue by Kelym** [pytorch/pytorch#11551]
## Issue description

When using CrossEntropyLoss, I assumed that as long as I ignore a target, its loss would not be calculated and will not get back propagated. Therefore I would pass logits = -float(inf) when I attempted to skip that target. 
However, even though the loss would be skipped; the loss.backward() would bring the infinity into my gradients. As soon as I change to logits = 0 for targets that I will ignore, the NaN problem went away. 

I think this behavior is at least worth noticing in the document for CrossEntropyLoss. 

## Code example

`
loss = nn.CrossEntropyLoss()
input = torch.randn(3, 5, requires_grad=True)
target = torch.empty(3).long().fill_(-1)
output = loss(input, target)
output.backward()
input.grad
`

## System Info

Collecting environment information...
PyTorch version: 0.4.1
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.5 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration:
GPU 0: GeForce GTX 1080 Ti
GPU 1: GeForce GTX 1060 6GB

Nvidia driver version: 396.44
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_static_v7.a

Versions of relevant libraries:
[pip] Could not collect
[conda] cuda90                    1.0                  h6433d27_0    pytorch
[conda] pytorch                   0.4.1           py36_cuda9.0.176_cudnn7.1.2_1    soumith
[conda] torchvision               0.1.9            py36h7584368_1    soumith

**Comment by Kelym** [comment]
Just a little bit more to the story, `0 * -float('inf')` yields `nan` while `0.xx * -float('inf')` yields `-inf`. Maybe this is the expected behavior, but it could really help to hint this somewhere in the doc. 
Thanks.
